### PR TITLE
feat: Use PropertyAccess Component syntax for nested variables access

### PR DESCRIPTION
**Pull request by TheoD02**
### Description
This pull request introduces a feature enhancement that allows the use of the [Property Access Component](https://symfony.com/doc/current/components/property_access.html) syntax for accessing nested variables within the context.

### Changes Made
- Added the capability to utilize Property Access Component syntax for retrieving variables from the context.
- Modified the `.gitignore` file to exclude the `.idea/` directory.
- Improved the example usage by replacing try/catch with the null coalescing operator for enhanced readability.
- Upgraded the `friendsofphp/php-cs-fixer` dependency to version 3.51.0 to align the local and CI environments.

### Example of Usage
```php
// Context
new Context([
    'nested' => [
        'key' => 'nested_value',
    ],
]);

// Getting nested values
$foobar = variable('[foo][bar][baz]', 'default value');

// Same as:
$foobar = $context['foo']['bar']['baz'] ?? 'default value';
```

### Warning
Not sure about the usage of the `str_starts_with('[')` condition for detecting when use PropertyAccess.

```php
if (str_starts_with($key, '[')) {
    return PropertyAccess::createPropertyAccessor()->getValue($context, $key) ?? $default;
}
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@
 /my-app.*
 /var/
 /vendor/
+.idea/
```
Comment (lyrixx) [inline .gitignore:8]: Could you revert that? It must be in your global git ignore 

https://stackoverflow.com/a/7335487/685587

File: src/ContextRegistry.php [hunk 2]
```diff
@@ -113,11 +115,11 @@ public function getVariable(string $key, mixed $default = null): mixed
     {
         $context = $this->getCurrentContext();
 
-        if (!isset($context[$key])) {
-            return $default;
+        if (str_starts_with($key, '[')) {
+            return PropertyAccess::createPropertyAccessor()->getValue($context, $key) ?? $default;
```
Comment (lyrixx) [inline src/ContextRegistry.php:119]: Indeed this is a bit weird. Could we find a way to always use property access?

As a user, I would expect:

```
$context = ['foo' => 'bar', 'nested' => ['a' => 'b']];
$foo = variable('bar');
$a = variable('nested['a']);
```

But When I write this, I really wonder why it's better than

```php
$a = variable('nested')['a']
```

(Note: I'm not against this PR. Just asking)

**Review by pyrech — COMMENTED** [review]
I'm not a huge fan of this PR as it adds complexity and a dependency for something already feasible in pure PHP (and so, a syntax more usual for PHP developers)

**Comment by TheoD02** [comment]
> I'm not a huge fan of this PR as it adds complexity and a dependency for something already feasible in pure PHP (and so, a syntax more usual for PHP developers)

That is true, this is mainly for the DX, WDYT about using this instead of PropertyAcces ?

```php
$array = [
    'nested' => [
        'key' => 'nested_value',
    ],
];

$value = variable('nested.key', 'default_value'); // 'nested_value'

$value = variable('nested.key.unknown', 'default_value'); // 'default_value'
```

File: src/ContextRegistry.php (hunk 2 above)
Comment (TheoD02) [reply src/ContextRegistry.php:119]: Yes, but the component syntax doesn't allow you to do that, as far as I know. An element of an array must be enclosed in square brackets like `[nested][a]`

The only thing that's better than syntax two (current), is that the default value must be managed again in the case, for example, that the index doesn't exist.

`$a = variable('nested')['a'] ?? null`

And worse if `null` is not the default value

`$a = variable('nested', 'another')['a'] ?? 'another'`

But as indicated in the comment here https://github.com/jolicode/castor/pull/302#issuecomment-1976336797, this approach could be handled in pure PHP without any deps

**Comment by lyrixx** [comment]
IMHO, you should not spend more time on this one. I don't really think the DX is better. I always feels awkward the "property access style" in tools.

**Comment by TheoD02** [comment]
Okay, thanks for your feedbacks and time 😃

**Pull request closed without merging**